lib/libc/common/include/string.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2024, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_COMMON_INCLUDE_STRING_H_
+#define ZEPHYR_LIB_LIBC_COMMON_INCLUDE_STRING_H_
+
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if _POSIX_C_SOURCE >= 200809L || defined(_ZEPHYR_SOURCE)
```

Comment:
You don't need to declare these functions when `_POSIX_C_SOURCE` is set -- the underlying C library will do that for you.